src/controller/AutoCommissioner.cpp

```diff
@@ -128,24 +137,39 @@ CommissioningStage AutoCommissioner::GetNextCommissioningStage(CommissioningStag
         // TODO(cecille): device attestation casues operational cert provisioinging to happen, This should be a separate stage.
         // For thread and wifi, this should go to network setup then enable. For on-network we can skip right to finding the
         // operational network because the provisioning of certificates will trigger the device to start operational advertising.
-        if (mParams.GetWiFiCredentials().HasValue())
-        {
-            return CommissioningStage::kWiFiNetworkSetup;
-        }
-        else if (mParams.GetThreadOperationalDataset().HasValue())
+        if (mNeedsNetworkSetup)
         {
-            return CommissioningStage::kThreadNetworkSetup;
+            if (mParams.GetWiFiCredentials().HasValue() &&
+                mNetworkTechnology.Has(
+                    chip::app::Clusters::NetworkCommissioning::NetworkCommissioningFeature::kWiFiNetworkInterface))
```

Comment:
Fwiw, all the "chip::" prefixes can go away.   This is all inside namespace chip.

Reply:
Will address in follow up PR - worried about breaking zap again on this change and blocking a bunch of PRs.